search-app/src/cmr/search/services/query_execution/granule_counts_results_feature.clj

```diff
@@ -111,9 +113,9 @@
   "Extracts a query to find the number of granules per collection in the results from a collection query
   coll-query - The collection query
   results - the results of the collection query"
-  [coll-query results]
+  [context coll-query results]
   (let [collection-ids (query-results->concept-ids results)
-        spatial-operator (extract-spatial-operator coll-query)
+        spatial-operator (extract-spatial-operator context)
         spatial-temp-conds (->> coll-query
```

Comment:
It is confusing that we extract spatial and temporal conditions out of the query, then extract spatial and temporal conditions separately from the `spatial-temp-conds`. Why don't we just extract spatial conditions and temporal conditions separately from the query and construct the final condition based on the spatial operator?